src/parse/decl.rs

```diff
@@ -835,7 +836,7 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
             }
             let (sc, quals, param_type, _) = self.declaration_specifiers()?;
             // true: allow abstract_declarators
-            let declarator = self.declarator(true)?;
+            let declarator = self.declarator(true, quals)?;
```

Comment:
Technically `int f(inline int g());` should be an error but I don't think we need to catch it until https://github.com/jyn514/rcc/issues/142 is fixed.